tests/test_schema_validation.py

```diff
@@ -326,6 +326,27 @@ def test_invalid_string_transforms():
     check_validation_errors(filename, expected_error_messages)
 
 
+def test_invalid_placeholder_answer_ids():
+    filename = 'schemas/invalid/test_invalid_placeholder_source_ids.json'
+
+    expected_error_messages = [
+        'Schema Integrity Error. Invalid answer id reference `answer4` for placeholder `simple_answer` (self-reference)',
+        'Schema Integrity Error. Invalid answer id reference `invalid-answer0` for placeholder `simple_answer`',
+        'Schema Integrity Error. Invalid answer id reference `invalid-answer1` for placeholder `answer1`',
+        'Schema Integrity Error. Invalid metadata reference `invalid-metadata-ref` for placeholder `simple_metadata`'
+    ]
+
+    check_validation_errors(filename, expected_error_messages)
+
+
+def test_placeholder_answer_ids():
+    filename = 'schemas/valid/test_placeholder_source_ids.json'
```

Comment:
we dont need to test valid schemas do we? it happens automatically?

Reply:
I'd forgotten about that. Yeah, it does.

Reply:
@iwootten It creates tests for all of the valid schemas here: https://github.com/ONSdigital/eq-schema-validator/blob/7d368204b584fa12cdd3c45d40d49cfa07ac1132/tests/conftest.py

I'm not sure how to make that more discoverable?